data-prepper-plugins/key-value-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/keyvalue/KeyValueProcessor.java

```diff
@@ -181,11 +197,16 @@ public Collection<Record<Event>> doExecute(final Collection<Record<Event>> recor
                     continue;
                 }
 
-                if (!excludeKeysSet.isEmpty() && excludeKeysSet.contains(key)) {
+                if (excludeKeysSet.contains(key)) {
                     LOG.debug(String.format("Key is being excluded: '%s'", key));
                     continue;
                 }
 
+                if (defaultKeysSet.contains(key)) {
+                    LOG.debug(String.format("Skipping already included default key-value pair: '%s'", key));
```

Comment:
You should not use a `String.format` inside a log statement.

There are a few reasons:
1. This formatted string now becomes a format string for the logger itself.
2. `String.format` string interpolation is slower than SLF4J's string interpolation.
3. The string must be formatted even if the debugger is turned off.

Change this to:

```
 LOG.debug("Skipping already included default key-value pair: '{}'", key);
```

Reply:
thanks for the insight! good to know. I will also change for the other logging statements for include and exclude key.